src/EditorFeatures/CSharpTest/Diagnostics/RemoveInKeyword/RemoveInKeywordCodeFixProviderTests.cs

```diff
@@ -0,0 +1,244 @@
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    void M(in int i, string s) { }
+    void N(int i, string s)
+    {
+        M(in i, in [|s|]);
+    }
+}",
+@"class Class
+{
+    void M(in int i, string s) { }
+    void N(int i, string s)
+    {
+        M(in i, s);
+    }
+}");
+        }
+
+        [Theory]
+        [InlineData("in    [|i|]", "i")]
+        [InlineData("  in  [|i|]", "  i")]
+        [InlineData("/* start */in [|i|]", "/* start */i")]
+        [InlineData("/* start */ in [|i|]", "/* start */ i")]
+        [InlineData(
+            "/* start */ in [|i|] /* end */",
+            "/* start */ i /* end */")]
+        [InlineData(
+            "/* start */ in /* middle */ [|i|] /* end */",
+            "/* start */ i /* end */")]
+        [InlineData(
+            "/* start */ in    /* middle */ [|i|] /* end */",
+            "/* start */ i /* end */")]
+        [InlineData(
+            "/* start */in /* middle */ [|i|] /* end */",
+            "/* start */i /* end */")]
+        public async Task TestRemoveInKeywordWithTrivia(string original, string expected)
+        {
+            await TestInRegularAndScript1Async(
+$@"class App
+{{
+    void M(int i) {{ }}
+    void N(int i)
+    {{
+        M({original});
+    }}
+
+}}",
+$@"class App
+{{
+    void M(int i) {{ }}
+    void N(int i)
+    {{
+        M({expected});
+    }}
+
+}}");
+        }
+
+        [Fact]
+        public async Task TestRemoveInKeywordFixAllInDocument1()
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    void M1(int i) { }
+    void M2(int i, string s) { }
+
+    void N1(int i)
+    {
+        M1(in {|FixAllInDocument:i|});
+    }
+
+    void N2(int i, string s)
+    {
+        M2(in i, in s);
+    }
+
+    void N3(int i, string s)
+    {
+        M1(in i);
+        M2(in i, in s);
+    }
+}",
+@"class Class
+{
+    void M1(int i) { }
+    void M2(int i, string s) { }
+
+    void N1(int i)
+    {
+        M1(i);
+    }
+
+    void N2(int i, string s)
+    {
+        M2(i, s);
+    }
+
+    void N3(int i, string s)
+    {
+        M1(i);
+        M2(i, s);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestRemoveInKeywordFixAllInDocument2()
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    void M1(int i) { }
+    void M2(in int i, string s) { }
+
+    void N1(int i)
+    {
+        M1(in {|FixAllInDocument:i|});
+    }
+
+    void N2(int i, string s)
+    {
+        M2(in i, in s);
+    }
+
+    void N3(int i, string s)
+    {
+        M1(in i);
+        M2(in i, in s);
+    }
+}",
+@"class Class
+{
+    void M1(int i) { }
+    void M2(in int i, string s) { }
+
+    void N1(int i)
+    {
+        M1(i);
+    }
+
+    void N2(int i, string s)
+    {
+        M2(in i, s);
+    }
+
+    void N3(int i, string s)
+    {
+        M1(i);
+        M2(in i, s);
+    }
+}");
+        }
```

Comment:
do we have negative tests showing we don't remove `in` when it should be kept around?

Reply:
Something like `M2` method in this case?
https://github.com/dotnet/roslyn/blob/4f7211b80ba975aea4d0d9504aa82d9fd138d1d3/src/EditorFeatures/CSharpTest/Diagnostics/RemoveInKeyword/RemoveInKeywordCodeFixProviderTests.cs#L200-L220
Or do you mean it would be nice to add more tests for that particular case?

Reply:
i'd like a test htat purely verifies the negative scenario and nothing else :)

Reply:
Added :)